src/routers/Questions/UpdateQuestionRouter.js

```
@@ -0,0 +1,43 @@
+// @flow
+import { Request } from 'express';
+import AppDevRouter from '../../utils/AppDevRouter';
+import QuestionsRepo from '../../repos/QuestionsRepo';
+import constants from '../../utils/constants';
+
+import type { APIQuestion } from '../APITypes';
+
+class UpdateQuestionRouter extends AppDevRouter<Object> {
+  constructor () {
+    super(constants.REQUEST_TYPES.PUT);
+  }
+
+  getPath (): string {
+    return '/questions/:id/';
+  }
+
+  async content (req: Request): Promise<{ node: APIQuestion }> {
+    const questionId = req.params.id;
+    var text = req.body.text;
+    var results = req.body.results;
+
+    const prevQuestion = await QuestionsRepo.getQuestionById(questionId);
+    if (!prevQuestion) throw new Error(`Poll with id ${questionId} was not found!`);
+
+    if (!text) text = prevQuestion.text;
+    if (!results) throw new Error(`No fields specified to update.`);
```

We only need to throw this error if the `text` and `results` field is null. Also use single quotes for 'No fields...' The other type of quote is used when we want to include a variable in the string.